src/lib/src/regsurf/sample_grid3d.cpp

```diff
@@ -46,12 +57,17 @@ sample_grid3d_layer(const size_t ncol,
                     const py::array_t<float> &zcornsv,
                     const py::array_t<int> &actnumsv,
                     const size_t klayer,
-                    const int option,
-                    const int activeonly)
+                    const int num_threads)
 {
     Logger logger(__func__);
     logger.debug("Sampling 3D grid layer to a regular surface...");
 
+    // clang-format off
+    #ifdef __linux__
+      if (num_threads > 0) omp_set_num_threads(num_threads);
+    #endif
+    // clang-format on
+
```

Comment:
You might consider defaulting `num_threads = -1` in the function signature instead. Passing 0 as the default number of threads initially alerted me since it's undefined behavior to do so, but I see it's just the "don't use" case... -1 might make it a bit clearer (at least in my head) and defaulting will also allow you not provide the argument unless it's explicitly being used for testing/benchmarking.